flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/classification/SVM.scala

```diff
@@ -242,8 +275,21 @@ object SVM{
         }
       }
 
-      override def predict(value: T, model: DenseVector): Double = {
-        value.asBreeze dot model.asBreeze
+      override def predict(value: T, model: DenseVector, predictParameters: ParameterMap):
+        Double = {
+        val thresholdOption = predictParameters.get(Threshold)
+
+        val rawValue = value.asBreeze dot model.asBreeze
+        // If the Threshold option has been reset, we will get back a Some(None) thresholdOption
+        // causing the exception when we try to get the value. In that case we just return the
+        // raw value
+        try {
+          val thresOptionValue = thresholdOption.get
+          if (rawValue > thresOptionValue) 1.0 else -1.0
+        }
+        catch {
+          case e: java.lang.ClassCastException => rawValue
+        }
```

Comment:
The problems is with how `get` is defined, if a value is not defined in the map, it will return the default value. So even if I clear a parameter, if its default value is not `None` (which is what we want here) `get` will return the default value.